### PR TITLE
fix(sequence): Preserve original refSeq on multiply-resubmitted ops

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

Fixes remaining known issues with 0x54e.

### CODE REVIEW

File: packages/dds/sequence/src/test/intervalRebasing.spec.ts [hunk 1]
```diff
@@ -16,28 +16,67 @@ import { IntervalStickiness } from "../intervals";
...
+	id: string,
+): Promise<Client> {
+	const { summary } = source.sharedString.getAttachSummary();
+
+	const dataStoreRuntime = new MockFluidDataStoreRuntime();
+	dataStoreRuntime.options = {
+		intervalStickinessEnabled: true,
+		mergeTreeEnableObliterate: true,
+	};
+	const factory = SharedString.getFactory();
```
Comment (Abe27342) [inline packages/dds/sequence/src/test/intervalRebasing.spec.ts:56]: i plan on letting #19398 merge first, which will include these changes.

Comment (Abe27342) [reply packages/dds/sequence/src/test/intervalRebasing.spec.ts]: updated.

File: packages/dds/sequence/src/test/intervalRebasing.spec.ts [hunk 2]
```diff
@@ -340,6 +340,30 @@ describe("interval rebasing", () => {
...
+		A.containerRuntime.connected = false;
+		containerRuntimeFactory.processAllMessages();
+		B.sharedString.insertText(0, "8");
+		A.containerRuntime.connected = true;
+		A.containerRuntime.connected = false;
+		B.sharedString.insertText(0, "J");
+		containerRuntimeFactory.processAllMessages();
+		A.containerRuntime.connected = true;
+		containerRuntimeFactory.processAllMessages();
+		assertConsistent(clients);
```
Comment (connorskees) [inline packages/dds/sequence/src/test/intervalRebasing.spec.ts:364]: I think this should be 
```suggestion
		assertConsistent([A, B]);
```

I'll think about a better test API for this sort of scenario.

**Review by connorskees — APPROVED** [review]
lgtm

**Pull request merged**